### ISSUE TITLE
Library folder tree statting performance enhancement

### ISSUE THREAD

**Issue by kevinvalk** [haiwen/seafile-client#725]
I have been noticing many stat fail messages in seafile.log (actual code in haiwen/seafile, but creating ticket here)
```
[09/21/15 23:30:29] repo-mgr.c(1899): Failed to stat AppData/Roaming/Mozilla/Firefox/Profiles/464549ow.default/sessionstore-backups/recovery.js.tmp: No such file or directory.
...
[09/21/15 23:31:29] repo-mgr.c(1899): Failed to stat AppData/Roaming/Microsoft/Windows/Recent/CustomDestinations/QYL06PJ55XG64QDBYOI4.temp: No such file or directory.
...
[09/21/15 23:32:07] repo-mgr.c(1899): Failed to stat AppData/Local/Temp/acro_rd_dir/FAP16C1.tmp: No such file or directory.
```
This is all normal behavior as these files are made and then removed. You can even test this behavior by creating a "New Folder" and then quickly giving it a name and hitting enter. What actually happens is that a "New Folder" is created and then renamed. This will also throw the same warning message.

HOWEVER, I have ```AppData\``` in my ignore file. This works nicely as AppData is not synced to the cloud. But I think it is a waste that seafile is statting through all these ignored files.

I think some performance improvements can be made, by including the seafile ignore paths in the repo-mgr.c code for the clients so that the code does not have to go through all the sub directories if a parent directory is ignored anyways. That would give a substantial speed increase for people like me that ignore massive directories (like AppData in my home folder).

**Comment by killing** [comment]
Yes, this can be improved. We'll do it in later version.

**Comment by killing** [comment]
This should have been fixed in the current version (4.4.2).
